crates/ark/tests/data_explorer.rs

```diff
@@ -70,24 +71,14 @@ use itertools::Itertools;
 use libr::R_GlobalEnv;
 use libr::Rf_eval;
 
-struct DataExplorerSocket {
-    socket: socket::comm::CommSocket,
-}
-
-impl Drop for DataExplorerSocket {
```

Comment:
When the Data Explorer is opened from a binding name, we'll need to make sure that at the end of the test they get deleted and we send a `EVENTS.console_prompt.emit(());`, otherwise there will be too many data explorer loops running at some point, which can slow down test execution.

Reply:
It would also be nice if these tests did not rely on global environment bindings and instead were self-contained.

Do you think we should preserve `DataExplorerSocket` in the mean time? I'd be happy to revert the revert commit.

Reply:
I can work on that in a follow up PR.